### ISSUE TITLE
Enable `object-curly-spacing` and `comma-dangle` on the ESLint codebase

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7725]
I think it would be nice to enable [`object-curly-spacing`](http://eslint.org/docs/rules/object-curly-spacing) and [`comma-dangle`](http://eslint.org/docs/rules/object-curly-spacing) on the ESLint codebase. A lot of our code is inconsistent about this (sometimes it's even inconsistent in the same object, e.g. `{ message: "foo"}`).

I ran the rules on the existing codebase to determine which options we should use:

```bash
$ eslint --rule 'comma-dangle:[error, never]' 'lib/**/*.js' 'tests/**/*.js' 'bin/**/*.js' --rulesdir lib/internal-rules/ | grep 'problems'
✖ 227 problems (227 errors, 0 warnings)
$ eslint --rule 'comma-dangle:[error, always-multiline]' 'lib/**/*.js' 'tests/**/*.js' 'bin/**/*.js' --rulesdir lib/internal-rules/ | grep 'problems'
✖ 13251 problems (13251 errors, 0 warnings)
```

It looks like we almost always omit the trailing comma, so I think we should enable `comma-dangle: never`.

```bash
$ eslint --rule 'object-curly-spacing: [error, always]' 'lib/**/*.js' 'tests/**/*.js' 'bin/**/*.js' --rulesdir lib/internal-rules/ | grep 'problems'
✖ 9272 problems (9272 errors, 0 warnings)
eslint --rule 'object-curly-spacing: [error, never]' 'lib/**/*.js' 'tests/**/*.js' 'bin/**/*.js' --rulesdir lib/internal-rules/ | grep 'problems'
✖ 16060 problems (16060 errors, 0 warnings)
```

It looks like we usually include curly spacing, so I think we should enable `object-curly-spacing: always`.

I'll probably create a PR to add these in the next few days if no one has any objections. I just wanted to create this issue first to give some advance notice in case anyone is against enabling these rules, since it seems like we generally end up merging PRs like that fairly quickly to avoid merge conflicts.

**Comment by platinumazure** [comment]
:+1: to `comma-dangle: "never", object-curly-spacing: "always"`.

**Comment by ilyavolodin** [comment]
I think our styleguide requires `comma-dangle: ["error", "never"]`.

**Comment by platinumazure** [comment]
@ilyavolodin I didn't see it when I took a quick look. What section should I be looking at?

**Comment by mysticatea** [comment]
Before, `comma-dangle` had been applied to our codebase, but it was dropped from `eslint:recommended` at 3.0.0.

It's fine `comma-dangle: ["error", "never"]` for implementations but I think `comma-dangle: ["error", "always-multiline"]` is better for tests because we can get better diff.
(though my personal preference is `always-multiline` at all :smiley:)

**Comment by not-an-aardvark** [comment]
I would be fine with either `comma-dangle: [error, never]` or `comma-dangle: [error, always-multiline]`

**Comment by vitorbal** [comment]
My personal preference for `comma-dangle` is also `always-multiline`, but I am 👍  for adding it either way, for consistency's sake.
I'm also 👍  for `object-curly-spacing: "always"`

**Comment by kaicataldo** [comment]
I don't have a huge preference for `comma-dangle`, but I do like the idea of nicer Git diffs :)

**Comment by kaicataldo** [comment]
Sounds like the only thing left to hash out here is how we want to configure `comma-dangle`. Seems like most of us would be fine `"never"` or `"always-multiline"`. Since we have one vote strictly in favor for `"never"`, I propose we go with `"never"` (which is what our codebase currently follows anyway).

**Comment by vitorbal** [comment]
Makes sense :+1:
